### DISCUSSION TITLE
feature: mdx-analyzer extra settings

### DISCUSSION THREAD

**Discussion by ajmasia** [LazyVim/LazyVim#4455] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs
- [ ] This is not a request for a new or existing extra (see above)

### Is your feature request related to a problem? Please describe.

Mason mdx-analyzer LSP don't recognize mdx files

### Describe the solution you'd like

Define an extra settings for LazyVim to set globally LSP, Format and Linting for mdx files (storybook)

### Describe alternatives you've considered

Manually settings explanation

**Comment by dpetka2001** [comment]
[main/.github/ISSUE_TEMPLATE/feature_request.yml](https://github.com/LazyVim/LazyVim/blob/main/.github/ISSUE_TEMPLATE/feature_request.yml?rgh-link-date=2024-09-03T10%3A21%3A21Z)

>    IMPORTANT: ideas or requests related to extras, should go in the [Extra Requests](https://github.com/LazyVim/LazyVim/discussions/categories/extra-requests)
    discussion forum.

You've also checked the box...
>This is not a request for a new or existing extra (see above)

**Comment by dadav** [comment]
This is my setup for now...maybe it helps you. No autocompletion tho :(

```lua
vim.filetype.add({ extension = { mdx = "mdx" } })
vim.treesitter.language.register("javascript", "mdx")
return {
  -- Diagnostics
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        mdx_analyzer = {},
      },
    },
  },
  -- Install additional tools
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        "mdx-analyzer",
      })
    end,
  },
}
```
